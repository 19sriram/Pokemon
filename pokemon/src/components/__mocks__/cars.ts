export const cars = [
  { id: 1, brand: 'BMW', model: 'X5' },
  { id: 2, brand: 'Audi', model: 'A4' },
  { id: 3, brand: 'BMW', model: 'M3' },
  { id: 3, brand: 'BMW', model: 'M4' },
];

export const brands = ['BMW', 'Audi', 'BMW', 'Mercedes', 'Audi'];

export const sections = [
  { id: 1, title: 'BMW', content: 'BMW Cars List' },
  { id: 2, title: 'Audi', content: 'Audi Cars List' },
];