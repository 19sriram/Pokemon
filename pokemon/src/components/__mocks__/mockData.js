export const pokemonMockData = {abilities:[{ability:{name:"overgrow",url:"https://pokeapi.co/api/v2/ability/65/",},is_hidden:false,slot:1,},{ability:{name:"chlorophyll",url:"https://pokeapi.co/api/v2/ability/34/",},is_hidden:true,slot:3,},],base_experience:142,forms:[{name:"ivysaur",url:"https://pokeapi.co/api/v2/pokemon-form/2/",},],game_indices:[],height:10,held_items:[],id:2,is_default:true,location_area_encounters:"https://pokeapi.co/api/v2/pokemon/2/encounters",name:"ivysaur",order:2,past_types:[],species:{name:"ivysaur",url:"https://pokeapi.co/api/v2/pokemon-species/2/",},sprites:{front_shiny:"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/2.png",front_shiny_female:null,other:{dream_world:{front_default:"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/2.svg",front_female:null,},"official-artwork":{front_default:"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/2.png",},},},weight:130,};
export const allPokemonData = {data:{count:50,next:"https://pokeapi.co/api/v2/pokemon?offset=50&limit=50",previous:null,results:[{name:"bulbasaur",url:"https://pokeapi.co/api/v2/pokemon/1/"},{name:"ivysaur",url:"https://pokeapi.co/api/v2/pokemon/2/"},{name:"venusaur",url:"https://pokeapi.co/api/v2/pokemon/3/"},{name:"charmander",url:"https://pokeapi.co/api/v2/pokemon/4/"},{name:"charmeleon",url:"https://pokeapi.co/api/v2/pokemon/5/"},{name:"charizard",url:"https://pokeapi.co/api/v2/pokemon/6/"},{name:"squirtle",url:"https://pokeapi.co/api/v2/pokemon/7/"},{name:"wartortle",url:"https://pokeapi.co/api/v2/pokemon/8/"},{name:"blastoise",url:"https://pokeapi.co/api/v2/pokemon/9/"},{name:"caterpie",url:"https://pokeapi.co/api/v2/pokemon/10/"},{name:"metapod",url:"https://pokeapi.co/api/v2/pokemon/11/"},{name:"butterfree",url:"https://pokeapi.co/api/v2/pokemon/12/"},{name:"weedle",url:"https://pokeapi.co/api/v2/pokemon/13/"},{name:"kakuna",url:"https://pokeapi.co/api/v2/pokemon/14/"},{name:"beedrill",url:"https://pokeapi.co/api/v2/pokemon/15/"},{name:"pidgey",url:"https://pokeapi.co/api/v2/pokemon/16/"},{name:"pidgeotto",url:"https://pokeapi.co/api/v2/pokemon/17/"},{name:"pidgeot",url:"https://pokeapi.co/api/v2/pokemon/18/"},{name:"rattata",url:"https://pokeapi.co/api/v2/pokemon/19/"},{name:"raticate",url:"https://pokeapi.co/api/v2/pokemon/20/"},{name:"spearow",url:"https://pokeapi.co/api/v2/pokemon/21/"},{name:"fearow",url:"https://pokeapi.co/api/v2/pokemon/22/"},{name:"ekans",url:"https://pokeapi.co/api/v2/pokemon/23/"},{name:"arbok",url:"https://pokeapi.co/api/v2/pokemon/24/"},{name:"pikachu",url:"https://pokeapi.co/api/v2/pokemon/25/"},{name:"raichu",url:"https://pokeapi.co/api/v2/pokemon/26/"},{name:"sandshrew",url:"https://pokeapi.co/api/v2/pokemon/27/"},{name:"sandslash",url:"https://pokeapi.co/api/v2/pokemon/28/"},{name:"nidoran-f",url:"https://pokeapi.co/api/v2/pokemon/29/"},{name:"nidorina",url:"https://pokeapi.co/api/v2/pokemon/30/"},{name:"nidoqueen",url:"https://pokeapi.co/api/v2/pokemon/31/"},{name:"nidoran-m",url:"https://pokeapi.co/api/v2/pokemon/32/"},{name:"nidorino",url:"https://pokeapi.co/api/v2/pokemon/33/"},{name:"nidoking",url:"https://pokeapi.co/api/v2/pokemon/34/"},{name:"clefairy",url:"https://pokeapi.co/api/v2/pokemon/35/"},{name:"clefable",url:"https://pokeapi.co/api/v2/pokemon/36/"},{name:"vulpix",url:"https://pokeapi.co/api/v2/pokemon/37/"},{name:"ninetales",url:"https://pokeapi.co/api/v2/pokemon/38/"},{name:"jigglypuff",url:"https://pokeapi.co/api/v2/pokemon/39/"},{name:"wigglytuff",url:"https://pokeapi.co/api/v2/pokemon/40/"},{name:"zubat",url:"https://pokeapi.co/api/v2/pokemon/41/"},{name:"golbat",url:"https://pokeapi.co/api/v2/pokemon/42/"},{name:"oddish",url:"https://pokeapi.co/api/v2/pokemon/43/"},{name:"gloom",url:"https://pokeapi.co/api/v2/pokemon/44/"},{name:"vileplume",url:"https://pokeapi.co/api/v2/pokemon/45/"},{name:"paras",url:"https://pokeapi.co/api/v2/pokemon/46/"},{name:"parasect",url:"https://pokeapi.co/api/v2/pokemon/47/"},{name:"venonat",url:"https://pokeapi.co/api/v2/pokemon/48/"},{name:"venomoth",url:"https://pokeapi.co/api/v2/pokemon/49/"},{name:"diglett",url:"https://pokeapi.co/api/v2/pokemon/50/"},],}};
