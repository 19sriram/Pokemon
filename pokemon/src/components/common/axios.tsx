import axios from "axios";
import { BASE_URL } from "./const";
/**
 * Base url for application
 */
export default axios.create({
  baseURL: BASE_URL
});
