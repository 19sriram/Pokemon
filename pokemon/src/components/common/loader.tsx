import './loader.css';
function Loader() {
    return (
        <div className="center-on-page">
  
        <div className="pokeball">
          <div className="pokeball__button"></div>
        </div>
        
      </div>
    )
}

export default Loader;