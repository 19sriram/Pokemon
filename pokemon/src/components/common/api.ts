import axios from "./axios";

/**
 *
 * @param url : Endpoint
 * @param limit : Number of list to be fetched
 * @param offset : Number to be next offset after first fetch
 * @returns : List of Pokemon with name and url {name: string, url: string}
 */
export const fetchPokemonData = async (
  url: string,
  limit?: number,
  offset?: number
) => {
  try {
    const resp = await axios
    .get(url, {
          params: {
            limit: limit,
            offset: offset,
          },
        })
      return resp;
  } catch (error){
    return error;
  }
};
