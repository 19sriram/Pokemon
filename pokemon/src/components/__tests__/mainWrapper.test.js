import { mount, shallow } from "enzyme";
import * as React from "react";
import { fetchPokemonData } from "../common/api";
import { allPokemonData } from "../__mocks__/mockData";
import mockAxios from "../__mocks__/axios";
import MainWrapper from "../mainWrapper";
import { act } from "react-dom/test-utils";

global.fetch = jest.fn(() => {
  Promise.resolve(() => {
    json: () => {
      Promise.resolve({
        allPokemonData,
      });
    };
  });
});

beforeEach(() => {
  let wrapper = shallow(<MainWrapper />);
});

afterEach(() => {
  jest.clearAllMocks();
});

const setState = jest.fn();
const useStateSpy = jest.spyOn(React, "useState");

let response;

it("executes async call as for default one", async () => {
  await act(async () => {
    await mockAxios.get.mockImplementationOnce(() =>
      Promise.resolve({ data: allPokemonData })
    );
    response = await fetchPokemonData("pokemon", 50);
    expect(mockAxios.get).toHaveBeenCalledTimes(1);
    expect(mockAxios.get).toHaveBeenCalledWith("pokemon", {
      params: { limit: 50, offset: undefined },
    });
    expect(response.data.data.count).toEqual(50);
    expect(response.data.data.results.length).toEqual(50);
    expect(response.data.data.next).toEqual(
      "https://pokeapi.co/api/v2/pokemon?offset=50&limit=50"
    );
    let wrapper = mount(<MainWrapper />);
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find("Button").at(1).props("disabled")).toBeTruthy();
    expect(wrapper.find("Button").at(0).props("disabled")).toBeTruthy();
    expect(wrapper.find("Button").at(1).props("children").children).toEqual(
      "Next"
    );
    expect(wrapper.find("Button").at(0).props("children").children).toEqual(
      "Prev"
    );
  });
});
