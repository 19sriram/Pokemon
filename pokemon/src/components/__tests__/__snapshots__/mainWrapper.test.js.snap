// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`executes async call as for default one 1`] = `"<div class=\\"mainWrapper\\"><div class=\\"container-fluid\\"><nav class=\\"controllers navbar navbar-expand-lg navbar-light bg-light sticky-top\\"><div class=\\"container\\"><span class=\\"navbar-brand\\"><div><img src=\\"https://upload.wikimedia.org/wikipedia/commons/thumb/9/98/International_Pok%C3%A9mon_logo.svg/1280px-International_Pok%C3%A9mon_logo.svg.png\\" width=\\"100px\\" alt=\\"pokemon-logo\\"></div></span><button aria-controls=\\"basic-navbar-nav\\" type=\\"button\\" aria-label=\\"Toggle navigation\\" class=\\"navbar-toggler collapsed\\"><span class=\\"navbar-toggler-icon\\"></span></button><div class=\\"navbar-collapse collapse\\" id=\\"basic-navbar-nav\\"></div><div class=\\"navbar-nav\\" style=\\"display: inline;\\"><button style=\\"margin-right: 1em;\\" disabled=\\"\\" type=\\"button\\" class=\\"btn btn-primary\\">Prev</button><button disabled=\\"\\" type=\\"button\\" class=\\"btn btn-danger\\">Next</button></div></div></nav><div class=\\"mainRow row\\"><div class=\\"side-drawer\\"><div class=\\"brokenWrapper\\"><img src=\\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAP0AAADHCAMAAADlCqUFAAACB1BMVEX/////zACAAAAAAAD/AAD/zgD/0AD/0gD/0QD/ywD/1AB/AAD/yQCEAAB3AAD/g4P0wwD/xQD/uwD/xgD5+fl1AABwAABhAADmuADr7O3crwD/wQBmAABqAACampr4xgCDZQDMogCHipGuiQAQCAAYDACTSQAYAAAAAAWhfgC/v7+ztbm5kgDc3d/T09MoFABUKgCsVgBUAABKNQB7foaSlZvBmQAAAB5ZXmmkpqsiIiJ4XADhswBvVADKysoyGQBFIgB6PQDwAADbAAClTFhpbXZKSkpfRwCUcwBGTFk2O0c6JwB5eXkAABNgYGCpqalFMAAtLS1rNQBdLgCeTwC6XABmMwATCQA9HgDJAABfTABOAABtWwBAAABbEAR1Xl4uAAAWHSmShl06QU8eFwBELgApMkQVHS0AFS7EoiuCbBweGQisjyQxIQBgVjEsLCyEc0OwkjOLgmuVfjdCQkKOZAAAACSregDjoQB3UADOjgBTIQBWMQBvRQD3rQA+FQDWlgCccgC2AACfPQCZIACySQWnZwigVw6tODiugIOmoJC1JwOzFhaVYWGNOQCIJQBnQkJcHx9bIjd8NkZJISffbXNsMjpnPCxlRB9oRkZlIgKPc3MiAACTgIA3NAAdIgBLRADgysrNqKiPODjRsLC4hISfXFz/1T//2WH/3n//55z/7Lf/9dUQ4r4CAAAXrElEQVR4nN1d+38TR5JHGnpm1LJ0g5GtGWFbGssvPQ1+CNuyBYYYyfgF2EnAgIG1eeyyvuQuEOIQkiMkm71LNhvChVxCgOzmdu+1f+R1z4ykeUozmtEM7PezP2w+QKu/XdVV1dXVNfv2/Z0hV0pP908sTG2fvraysjIyMrKycu3a6cWFib61dD7n9ezahUg+MzF1miDGi5VUlo/NsnESSiADDBtLESJOLfel/54WIZLLTCzODJcr2RgLIKRomiQBgq8G/F/wV4QMp5bXchEPp8zlkIYuTC1un0bY3l6cmpiczuQ4i1PiMhPbO9crWTYOIU0qKKsAYoQap/raw63xjEvTCzdmlsoJrKEsI4LlC8lUYnxp5sbCdIkzNU5ueuHmUiLLBgTehrSr7GdHNPSX28xUDW564dbuXJJnAhRSUVFDRZAIFKQDDJ+aWzo6Nd14b3KZhaPjFT5OUSaICyB/rSE/4xJpAZH0xNElNGMSksZTxssAaR9fGSempg10IN9/azTBB9DqmSMuAP5Gw9494xfJLBLFQhyamzEgaUjyleGbk5oZpheIcpaBlBXmwqreVpNfc5F6ImZ1xmgF2NT1lf6cYpwiD6BF5sJgWfW2X3WHe36BKGLqlmeMdQAyyWOnhS0QyUyhJaSsCr060I6K/Nn2kMWubBJFW8iDobAisnZtvOCDrVCvLQBbGZ7KrO4kYiTVCnMM+rdqvTfnVywRL00uzuyOF1PJbKGAPFjx+jXiDNuKoioXgCL5MpECsOVxALOiIp93mHpp4ibyZbPIpGNXhkFTMM7zrYtdBqwARDHeKn34jyryGSeZ4z25mWVojfdF/ss+dXEkKpCMt/hvKbXeO2nu09gc0a2ZIwsArS4kyarsvXMRbm6CmONbs+guAfh22kQ+vT2aDdiw6C4AUJttUXtu8mY5ZtuitxswoSSfdoR7emqwwrzaYseAv1VueidcXX5iZI4HLQcf7gEmFeTv2A9yuMlb46/6bpcAleG9/QN95sZginnld7sIFXm7W57rJ4qzkH4tqCPyKTn5VZuJvPwUkQq0Hm27DAArMu4zNgWfuTHOU6+L2FGE55P7eXsRTqR/pMi+FoZOAs0syaydLVMf6SdS8ddG5X1Y6wv1Ld9vm7vv9VF5H9b6YpX6O2u2jN3rxx1AvnquWbUZ2iHu4DXjzmyKWr9q179niMrrEM/WgNOgwqlmZjlt954ud7rMvC62DgCSpqg4X8TM1+xfVEQmBvlXnzu+/KEpSMXZQmpu6driZNqRbO00kXIsKdcuINYMG0umimdGZ7ZNX382B3djjnnlNzyVGB+eWZzqz6SdvYyfJl4DpfeRhZtOkpbAbZcDr7rSC4BlRxPzAjJEwSnBC/fy9sfClk1vHMDecph7ZLUcp23PVwBJ+1i+wDOt37+JHCnAZ1PJAqvNm8Mzzl7C50ZuOyR4QPLvifHmUoqhWh+HZhMD4jjjWVrFn05NOkk+QxC70B5rCST7bn4fV1oVJ55sVf+B758mF2vHtTMx5TIC5oaD5JfRL+w6ovc0LyUUSuK8N03W16jJd7wbwfeEZ6v8C0rZUEuOXUVzd/D4I4wDmk/y09VR85Latkb/V5In76vRVwiHKjpzO1GbJsE6YKSZ/vq4p8Rhiy3sKKpSC2Nq9BXTI5MObfxMdfikfdWH78oGXq5KzXIMQfL183mkOr1x+fRAbNER8rW1JcZtGGhpTnxJNvI70ri7li0f/c+yYWaq85MXRADmmhPkl2vkiZGCXeHTctHXlzVrUfgkL/Pm+dowcuED344DIf4qIYNlu6coLAgEACsLQNfq425aVCryPdkM79THke98aN/oc2cJBZbi5qUESIqMyZQxgNgXamWWpVPycQMWSwzrV+2cjDwKHmTsy3ZvZbkZQoVd1pyBBiQEfJG4MVeTKiIfICsEcXa1r6//lGpYSyVMSIWOSrWa+WXFMHOyydEVmwcdTs0dK39KHVTqMKcgmywTC5l9paKcfQcoEm/rjGnN6iMVEorKZ85qZCM3y3RqujnDBsjpTpTYydJGhxOAiVOx5BxxelLQu+lKzRAh8gd948Ojo4ODAwPqMVPmzSlaRV9Bf2bEgGwYmw4/b/ATCBVWVQeND6wUJMFsNjFOLE6WquZ2sr4TEfsDHWeOnT9/bHTwbTX9ihX2HYEkMTCgXUIEOfvCgg3yJWPySP9fiDXwYukhJONsIZko7x7d7s8oTpb9dZVGoj9wYO744cPHMf/63E/1LSvNVVPRxw8WiMFBtIQa/jtylzc7ZYP9qnZbqTCeyPLZZKVYPrN7bXtiMpPTupipGKhP++CBQ3cPnzt3bv3w+br47+zDjt/8vg/44nGGJ0aPqbfQWukd4mfZvgeszWAvwuVK6bW1vuXl1VOnzs7MyH5rYObs2TuLUwv909PpPGfoWd+vuzIk+0OH7q3PD83Pnzt8vCZ+HPbnLJwgkOjjh2LE+eOCCtX59+OtWpSzZ5yuto7UYPLvD8rd/cEDnR/cHxraGBqaf6sm/hk01PKgeZMvsO+8fnh9fR3todGa/mMjuyJXIRC/7TB7q8iVoWzaHQc6Oz8a2jhy5MiQIH5J+5fvEAnzsR4aBrFHOwip0FuCDon8S/i4owhFqUEPn5ZhZGS0MPtDnffmN468+eaRjaH5dUH7RbW1cHQW2X9wdX4IqdB8lT8ynsi9zikWEY46X2pvCf3y4wtS/UOdnceHEHvEX9J+gb6lAz5if6Dz4715pETSHhJtyJ1VYlaxiPZDXZu4IReqKPwP1zfU9AcsldeLw3xwHLF/U9hD61UTmlAGDVTCqexOa4jsAMW0sd3rvHuuRh/P+22LQb7g8JHw760LSoT3kLSKuwHlX7Qb6tpFvgyV8+7Aui/RP7KBTN/50QGiQKumbUr4H999a6NKH68iMciiP5H/PTI74Sn7aWX4HhDn3XnvPjb8G9jwnx8e5MmAZfYHBfrCMNVxdtiOuHIge8GefSzGgGreIv0P9u7PI5xb/2Q4cQCffHVZkkYvG7DZx/Q//OjcEMb8W8eJBMvE1ewZV1+UqhEZ9GnZC/TRtt178NHeXrbz4MEOA/YUn8gahAFV+miUT+7fv//J8bsfdDJMXD0QJLx0+KptX+V/UOCPcAjhgBF7kicMj34BiT7i//EHH/7Lxx+jsQTRq9h76vImtSe3gCB+dNg7dEjkjsn79NhTlxD7JUPdF0Y5JC0jWkdMXj2Ot0Z/WyeEC0j8RXR0dBhtexp3hLhkFAELg8Srq3jgQDwe7/CpBwIxD81eRP/sEhDRIfwvYGjzAEjsbjZIIUurGMeLGBcXUT0QCBDesS/NGQWwwkzxZI3JC89MG96YS4soqY/uOHDJuzYiE1bvKKwjUIPuH9tO69rAaQcuPe2B5O2k9myBG64rvlA46ESdjjUA5qhX7Otne5JiCsmHD5MFxu13qd7FOwvS0Y2kC48+PbEf4cSnj5JtrfLEtwlQ/jSKmis1n2hbsCK6K8g/EqiLOPEZ70zxjx53yBTuHfn8yL1Y7bbJsRIGq+AIgSZMXtivwMlUm+pcKfZ3F8R1PvkoJi2xUyUMlpEWssuwcmK/CifutkP6QLHMJ5PSb5Aj3rDvx0E+lTypJo/op5ynD8gvFMt8QjoewmFvMpv4bA/YT7Xk9++/EHPc9MEh9W+IwQYsexPtjQSAD36hR37//q9t1/+oyT/U/MYXgoJRCW+iPWT0AHtBhzrels70W6kBzGo3mCh8MtvffKrOIzcO0SFVn/z+/Q+dFT7cUA5/4gKy/kJ5nUexLi7agI+M2H/pqOwBqxT97/Mcl//XB7jECLDbHrEH4Esj9p+2VpxqADqplLyYz4pkkh0kiDtdp28KmQQNGF2LX9+UTkGlYyerTi7ybzwFPDnnTKcasT856yh7pY6dqF9g5ZLQG/YVGsQN2X/q5GFX/Ttf9dUPdpH3PEluYatHu7TvSQX733P75B0F7LaVaAn5OcpH/c6I/YajHk+h+b/gNwty+l50h82dgcjZ6kT5ws50NuMHj9SH/oPw65ysaVTEC/o41gOfGSh+y/3wdEFma8v6TPr1vCyrwXlwjY8LF8iCrvBPOPDKQY6aczlRF7O8JbYzz48tQTjhao5eAr52Or0pHXJOypukrckou6/7eRToo2P3H7XkP3PicZMCgMRb7IJSw2WUOfcPercwScB8rSb/R9b5xCZgvtr/lfokL6Ofcd3t9VWwWwP0XcXeP/lFW7K6JBPVqLfc8LvVHrmGCCFYdgBjQxeqWacLj9rzfh2AvXD3c/UMcvWt4L7hm0yI+TuSYpMbn3/55ecbSbbpy4aWQDNvBf3+rivqGZTq1QvuG76b1R0OSAgBgO1qwQb5MT9G8Bv19s7URO6+8EvHZK7NdGQvdsw32/0eb61UT8gv0n+s5lg78ETcN/sTCYu5a0BDyMR4jNk4xJ+6aBoakPEHXf4axjT0q//HdbuHAr6sFfoAspXxb5+8IeDJk2//ffxSpRADjbqjg6rWSwiNqfxeLcq33WPIOiJHefNBLfBdevLdPyjx3RtvfDueyLKkfsdK0ne3qvVG9KuRTsmDxD5HxEzTp8tq7tUl+O7JsUtZbYSIBL8V9KugoZ8X/Z4Hwb5A3+RZHsRGv3/6RNL7qvY//f77/0AoFxMJVd0n3id7KsGL9KMqnhnB73mw7xG4lYLJvQ9AnGFYVjR6sViMZVmGifto/FUjSt25EtDxSlQjeBGPVVu8P53J2Gum1zLSu0sps/Fd/RMx9U/G6P9FypfcCutzR47vB+UUMvu8quAooXh3c7PlHu+6ICFTGTPkjhBWBL0euHoJeSJOonCEmHXq0hqQFMXvGel8Fb3yw64XmS2JPIOlDmOEae1vQBwFzCRTuDvW3Zg6tnxjdV3v80rtOUKK9Enf3JLJp+uGIofxWPLuj9FwE7FLqJ54PMlpir98sxbrINdMVKw9wZAxpymm8HB+LNrb29ttkn23uNlL3jg6jEV5iQoZSLSSyUXa7os9PBftiT59vlbiuFz6WXPFF3Qfz2DNuwL9CeUzuxYasOJWoNm9rd6u6DNZqLamZ+5D6rgn3Lcv59mWR172us3jPKBAYQ/t867HylCF69UT9cWwKuDvWfPwOR5H2LuyIOFsZaw7GOp6qi63TOvIPnT10u2fFPyDV7x8h3rLTm0OoMnsOnbqwahGgJEejdIjDLMwO3CxRj/Ue8w7e9dCYkPOnYqntoSUhTZTuW/fNyqj3/3TMALSNBpcIoJ4JYLB0BhPD3un+OljLZNHZ7eHUhzboyXAPVV7vPALdDRicTEIgMmRi5eHidvE1iwFmDZ87cwcVC/hLZGP3a3GsT3ahERarfZoc7ysn4Ygn8rGWHKzQOE+fk72F7SChWTLF/TgclW43RpnzV3R8fUh+UqT+CuoZOES1jyY8KRWb19pqfXqBDopmfSges9zz3t14jxs8NR5D5/QuQqQw54EO0dtFCUBn5Sm7FJOPfesR8s9GO7tiWobtpC84G1JdsUD8v0VOwcaKiUKPyw7n+T6HnfrxfddxOalREWd80K8JTNQcV/3c4St+3nARKV9/xxLn8uvPYvqqbyg9gOzsNHXWmj37f6i9UaRCsCHEtdgdzQa7ekOGp/qQtFLDdWMdN3u55ds5nHIgqlTnEi/STzdjn7aDXHD7tsD6oF58qHLjWuAUMzj6jkvPW5T9GA2app89PLwQOO1ptw1fO/bLUCGd81lbwS9L/Bsk7UMuGn40tp+C9ZQ8/dm2I80/yAdnXKx88oNu9XXZLJRpl4t++a+FQD3hG/b4PvIVFTncs6A/bCJJAKdcm3nT1n09TqigwULVq+xuxd/Iu6W2ecIazWoQM9b0+yPpu2emd+DCZeaj/Rb6I3qw8eQst7fVxajNBT+TyZUH7AuBXw3LSU1SHZrTHe1AKyYo29K9X1w3JWTLi7Kt0Q+dNGgig1eNGf6QoSJ2i6X2o31m28MK5L3dz3IMnoLALdMqv5VEw8dkN1zg/01C4ovkEcIj51LaeuWodlYv+uFGdUvu9CCwYriV8kj8fmDFzWrRpkNd025fFde406at/h18hhBTTLIPPueTRMun23HZ7NUWDQd5SrJ+/1jauHTKbMuP2SmIasbvVfe1gvd5Jyrf4zIKykE1d0oyKxp9mMmnJ4LvVeabXuSl6SkIY+ErxIgYLfCJsP9ENFc5ehK28O9TOM28MjviBR1yPv9e6qVA3TK5Ek3FG2+88lC2z1+f2PXSyVSlCF5f1QtQACZdbPCb2r2Xeg8M6XNqytmsMMABflguKurXoXzQBP01HK7zdj3vmjmawB7ut3sFxsaX8DsQDn5YPTZWjqz9uyxlKvvfpBkSUUtOpk1mdsNXW3WwwYwbXd57zSM9EB8AOK3U1uSvJ5XfVD+Sq9o38LRHx/y8XquCrDaC1t9BIkmr9wA0/YWDDONU+vkLguqkg/1yCsr0rVCnGB3dC/LVF/zxM2mOUK9Lxtfm7qw7wcaf/KHzI5DSfL+bmVZyWO5HMNj1UZk8JxJ9v7Q5cZXhyTfdpvfrBU6LG7+KOp4UPl2jFPKOBSVatpNh7vY7jd8A+PwdyL10HjfYzYJQgxhVO8H1CVooajYJIy2kN4NEo3ewLjQXrSxzRcmUSAuhtCZTlU7m1YnckI9KQgpkopZuNALE6whfRBv/0X+QvOzJh1IDHdrno481z6u6Z7fe5gs8Kazu9jyGb+BoVxIaus0jtZKAb6MhtQvRn/QVeVguLvXvOYj+uu3Dfo2goALR7ySiUsseOlyKKyqI4xYkHADhLPMgP4rADjnQko7QjQ9a5HJgZCmEC1nNqhpjChDxjfHtd9nB7DiSjvtxaY39/Al0uXHqn+mW3NtGThFAJBZTZHK8wJJFd1pNJY2bJxdBZWIhvxPVf9Md9tbRlTwt3Q8QaQCUHrDDACN1sOta7yVZh6fTF4O+VVGLzJm+tayAaqpQXQwThDl7GyARDoQiFUGp1x7gTrd7LM/IDYcCj5T/qO0eadujNBYLasGKJJPvCB2z5zZeX9x2s2arZtNdj4AREhdg6n19i2gW/GtWvyhQoZhki5X6+UbXiwBmoY/d6s03xHFDz7QKB0ApOtfjJls4PNBPJVK/RxVhXpOWPyQJiMu/J77X4xZmDN8dggY4uLFsd6QX9EfwgnRR/VPOND93noLZaNaGsQ+JDydCkWf1QzxFfvc/V0Gh3s47n7HgclhVr9PODk7XJNzz9iz52trfc/GHDB5OpteBO1FK+HS0ZSmnxJ+Uw75q3UtR+dcDCdc/Y9GWRXSbaMvYmI4q4w3YfEFwkt5SULP02+eOhLgbzFGXtar7+VwC8MpRpadBUQYo67mwR/wlsz9YFvxQ9FZwxADMNc8YY/4T14rJ1mpYQ6Q7XgR1e4IdmN8/PTMiLwnRr+KicpPxMtEMsZACHF4L0e0aoxzOq9KrZFvlMqG1z37WhDx0B8KRi//RBAvN1/0KtnXD7m23H0T8h5+GTO/g22a8Eo02KvO3jjDvhl5NxLZBuiXuTe/mmIt2OVsmP1QtAl5Ny4xDED0NpJq9aRjI9RrTt67T8RxREOVDgvvw7krNo440eZNnDz7UlTmauMNHYx+c+WbZo1jGmHL+PKiDq9c3oLJUtOWyRtGeAr2Hn0whjjXTvbB9bipmliPvguaK5u/f7WO8AOTfSc9cvjTqawjWXpddBfNfmrLI4c/xQNHEtU6CPVWTHetIguenHFXak+pHScfTVqohfYk3MmdQbq51Q7h1wo7TMGbL8RlEjRad2fuZpXkx8w2qhTZt79WSwcL+GGahSelpnHRYgN2yovv493CqTbndT9431SMI4MXwR63K0ShJN/jd1L8pt28jP2o+8FepkgKXyeGv+nyB8NiSk+CDfJde9bblnnxdUAu97eIVCkT4bjcn/7zz3/+y1/++ssvv9hZi+5EC58Tpea867CnQQQtxp/QYqC1QIsRlNYibGItrMQ4cvae5baaIoIWA60FWoy/osUIB8NdhmuBK/isc8ehrktvcO0Dr4WgGH+QrYWwGP4eFOAFTKBDBZ831zkOQL4Yv/4vjP8W8T91/G8N/yfib2q0ZvP/H3BotZQDUxPTAAAAAElFTkSuQmCC\\" alt=\\"something is broken\\"><h2>Something is broken</h2></div></div><div id=\\"backdrop\\" style=\\"display: none;\\"></div></div></div></div>"`;
